homeassistant/components/airvisual_pro/__init__.py

```diff
@@ -0,0 +1,106 @@
+"""The AirVisual Pro integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from pyairvisual import NodeSamba
+from pyairvisual.node import NodeProError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN, LOGGER
+
+PLATFORMS = [Platform.SENSOR]
+
+DEFAULT_UPDATE_INTERVAL = timedelta(minutes=1)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up AirVisual Pro from a config entry."""
+
+    async def async_get_data() -> dict[str, Any]:
+        """Get data from the device."""
+        try:
+            async with NodeSamba(
```

Comment:
Unrelated to this PR since its a break out, but usually Samba connections are expensive to build and tear down. Ideally we would hold the connection for a long time instead.

May not be possible with this setup though

Reply:
Good call; will follow up.

Reply:
https://github.com/home-assistant/core/pull/83869